docs/source/concepts/ledger-model/ledger-integrity.rst

```diff
@@ -226,7 +229,103 @@ becomes apparent.
 `A`'s commit is not conformant to the contract model, as the model does
 not contain the top-level action she is trying to commit.
 
-.. _`da-model-authorization`:
+.. _da-model-key-consistency:
+
+Key Consistency
++++++++++++++++
+
+Contract keys introduce two validity constraints for ledgers: :ref:`key uniqueness <def-key-uniqueness>` and :ref:`key assertion validity <def-key-assertion-validity>`.
+To capture these notions, the contract model must specify for every contract in the system whether the contract has a key and, if so, the key.
+Every contract can have at most one key.
+Additionally, the model must provide the function `maintainers` that takes a key and returns the non-empty set of maintainers of the key.
+
+Key uniqueness ensures that every contract key refers to at most one contract.
+It ensures that a contract with an associated key can only be created if there is no active contract with the same key.
+
+.. _def-key-uniqueness:
+
+Definition »key uniqueness«
+    A ledger is said to satisfy **key uniqueness** for a key `k` if every **Create** action on a contract with associated key `k` happens only when there is no active contract with associated key `k`.
+
+    The ledger satisfies **key uniqueness** if it satisfies key uniqueness for all keys.
+
+For example, suppose that the painter `P` has made a paint offer to `A` with reference number `P123`, but `A` has not yet accepted it.
+When `P` tries to create another paint offer to `David` with the same reference number `P123`,
+then this creation action would violate key uniqueness.
+The following ledger violates key uniqueness.
+
+.. figure:: ./images/double-key-creation-highlighted.svg
+   :align: center
+   :name: double-key-creation
+
+.. _def-key-assertion-validity:
+
+Definition »key assertion validity«
+  A ledger is said to satisfy **key assertion validity** for a key `k` if and only if at every key assertion **NoKey** `k`, there is no active contract with key `k`.
+  
+  The ledger satisfies **key assertion validity** if it satisfies key assertion validity for all keys.
+
+Key assertions can be used in workflows to evidence the inexistence of a certain kind of contract.
+For example, consider a subscription service that operates a support hotline on the ledger.
```

Comment:
The problem with "handing out anything unique", be it accounts or phone numbers, is that those typically entail the creation of a contract with the unique thing as the key. But then, you don't need a `NoKey` assertions because that's baked into the `Create`.

Reply:
As discussed, changed to blacklisted customers.